silver_braintree/payment_processors.py

```diff
@@ -370,9 +373,59 @@ def execute_transaction(self, transaction):
 
         return self._charge_transaction(transaction)
 
+    def find_lost_transaction_id(self, transaction):
+        if transaction.data.get('braintree_id'):
+            return True
+
+        search_result = braintree.Transaction.search(
+            braintree.TransactionSearch.amount.is_equal(transaction.amount),
+            braintree.TransactionSearch.payment_method_token.is_equal(
+                transaction.payment_method.token
+            ),
+            braintree.TransactionSearch.created_at.between(
+                dateutil.parser.parse(transaction.data['requested_at']) - timedelta(seconds=1),
+                transaction.created_at + timedelta(seconds=61)
+            )
+        )
+
+        transactions = list(search_result.items)
+
+        def sync_transaction_id(unsynced_transaction, transaction_list):
+            """
+            :param unsynced_transaction: A transaction with missing braintree_id.
+            :param transaction_list: A list of braintree Transaction response objects.
+            :return: True if the transaction was found and synced, False otherwise.
+            """
+            if len(transaction_list) == 1:
+                unsynced_transaction.data['braintree_id'] = transaction_list[0].id
+                unsynced_transaction.external_reference = transaction_list[0].id
```

Comment:
The save will be done in the `_update_transaction_status` method. I just set the braintree_id so that the update process can continue.